File: Java/src/main/java/br/com/locaweb/locamail/api/model/Agenda.java
package br.com.locaweb.locamail.api.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDate;

@Entity
@Table(name = "T_LCW_AGENDA")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode
public class Agenda {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_agenda")
    private Long id_agenda;

    @Column(name = "id_email", nullable = false)
    private Long id_email;

    @Column(name = "nome", nullable = false)
    private String nome;

    @Column(name = "descritivo", nullable = false)
    private String descritivo;

    @Column(name = "id_usuario", nullable = false)
    private Long id_usuario;

    @Column(name = "cor", nullable = false)
    private Integer cor;

    @Column(name = "localizacao", nullable = false)
    private String localizacao;

    @Column(name = "notificacao", nullable = false)
    private Integer notificacao;

    @Column(name = "horario", nullable = false)
    private String horario;

    @Column(name = "data", nullable = false)
    private String data = LocalDate.now().toString();

    @Column(name = "proprietario", nullable = false)
    private String proprietario;

    @Column(name = "evento", nullable = false)
    private Boolean evento;

    @Column(name = "tarefa", nullable = false)
    private Boolean tarefa;

    @Column(name = "repeticao", nullable = false)
    private Integer repeticao;

    @Column(name = "grupo_repeticao", nullable = false)
    private Integer grupo_repeticao;

    @Column(name = "visivel", nullable = false)
    private Boolean visivel;
}

File: Kotlin/app/src/main/java/br/com/fiap/locawebmailapp/model/Agenda.kt
package br.com.fiap.locawebmailapp.model

import androidx.compose.runtime.Immutable
import androidx.compose.runtime.Stable
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey
import java.time.LocalDate

@Immutable
@Entity(tableName = "T_LCW_AGENDA",
    foreignKeys =
    [ForeignKey(
        entity = Usuario::class,
        childColumns = ["id_usuario"],
        parentColumns = ["id_usuario"]
    ),
        ForeignKey(
            entity = Email::class,
            childColumns = ["id_email"],
            parentColumns = ["id_email"]
        )
    ])
data class Agenda(
    @PrimaryKey(autoGenerate = true) var id_agenda: Long = 0,
    var nome: String = "",
    var descritivo: String = "",
    var id_usuario: Long = 0,
    var cor: Int = 0,
    var localizacao: String = "",
    var notificacao: Int = 0,
    var horario: String = "",
    var data: String = "${LocalDate.now()}",
    var proprietario: String = "",
    var evento: Boolean = false,
    var tarefa: Boolean = false,
    var repeticao: Int = 0,
    var grupo_repeticao: Int = 0,
    var visivel: Boolean = true,
    var id_email: Long? = null
)

@Immutable
data class AgendaCor(
    var cor: Int = 0
)

@Immutable
data class AgendaGrupoRepeticao(
    var grupo_repeticao: Int = 0
)